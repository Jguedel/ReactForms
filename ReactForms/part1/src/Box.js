const Box = (props) => {
  return (
    <div
      className="singleBox"
      style={{
        backgroundColor: props.color,
        width: props.width,
        height: props.height,
      }}
    ></div>
  );
};
export default Box;
