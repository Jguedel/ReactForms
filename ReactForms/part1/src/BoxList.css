.boxContainer {
  max-width: 1000px;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}
.singleBox {
  width: 1000px;
  margin: 5px 5px 0px 5px;
}
.boxForm {
  margin-bottom: 10px;
}
.boxForm label {
  padding-right: 5px;
  font-size: 20px;
}
.boxForm button {
  font-size: 25px;
  margin: 5px;
}
