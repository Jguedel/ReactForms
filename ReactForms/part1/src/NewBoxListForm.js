import React, { useState } from "react";
import { v4 as uuid } from "uuid";
const NewBoxListForm = (props) => {
  const INITIAL = {
    width: "",
    height: "",
    color: "purple",
  };
  const [formData, setformData] = useState(INITIAL);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setformData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    props.addBox(formData.width, formData.height, formData.color, uuid());
    setformData(INITIAL);
  };
  return (
    <div className="boxForm">
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor="width">Width:</label>
          <input
            className="width"
            name="width"
            type="number"
            placeholder="width"
            value={formData.width}
            onChange={handleChange}
            min={20}
            max={300}
            required
            step={20}
          />
        </div>
        <div>
          <label htmlFor="height">Height:</label>
          <input
            className="height"
            name="height"
            type="number"
            placeholder="height"
            value={formData.height}
            onChange={handleChange}
            min={20}
            max={300}
            required
            step={20}
          />
        </div>
        <div>
          <label htmlFor="color">Color:</label>
          <select
            className="color"
            id="color"
            name="color"
            value={formData.color}
            onChange={handleChange}
          >
            <option value="purple" select="true">
              purple
            </option>
            <option value="pink">pink</option>
            <option value="aqua">aqua</option>
            <option value="BurlyWood">BurlyWood</option>
            <option value="FireBrick">FireBrick</option>
            <option value="Gold">Gold</option>
            <option value="DarkOliveGreen">DarkOliveGreen</option>
            <option value="DarkSalmon">DarkSalmon</option>
          </select>
        </div>
        <button className="subBtn">Add!</button>
      </form>
    </div>
  );
};
export default NewBoxListForm;
