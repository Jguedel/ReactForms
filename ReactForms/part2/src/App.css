.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.item {
  margin: 5px auto 5px auto;
  max-width: 500px;
  outline: solid 2px black;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  text-align: center;
}
.delBtn {
  background-color: red;
  width: 25px;
  height: 25px;
  align-self: center;
  margin: auto 10px auto 10px;
  text-align: center;
  border-radius: 8px;
}
.delBtn:hover {
  background-color: rgb(112, 0, 0);
  cursor: pointer;
}
.taskText {
  text-decoration: underline;
  font-weight: bold;
  font-size: 20px;
}

.todoForm {
  display: flex;
  justify-content: center;
}
.subBtn {
  background-color: rgb(165, 164, 164);
  margin-left: 10px;
  border-radius: 8px;
}
.subBtn:hover {
  background-color: rgb(46, 46, 46);
  cursor: pointer;
}
