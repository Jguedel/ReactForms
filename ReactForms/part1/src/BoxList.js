import React, { useState } from "react";
import Box from "./Box";
import NewBoxListForm from "./NewBoxListForm";
import "./BoxList.css";

const BoxList = () => {
  const lists = [{}];
  const [items, setItems] = useState(lists);
  const addBox = (width, height, color, id) => {
    setItems((items) => [...items, { width, height, color, id }]);
  };

  return (
    <div className="boxList">
      <NewBoxListForm addBox={addBox} key={1} />
      <div className="boxContainer">
        {items.map((box) => {
          return (
            <Box
              key={box.id}
              width={`${box.width}px`}
              height={`${box.height}px`}
              color={box.color}
            />
          );
        })}
      </div>
    </div>
  );
};
export default BoxList;
