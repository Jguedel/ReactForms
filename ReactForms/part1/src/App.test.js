import { render, fireEvent, userEvent } from "@testing-library/react";
import App from "./App";
import Box from "./Box";
import NewBoxListForm from "./NewBoxListForm";
import BoxList from "./BoxList";

//BoxList render
test("App can render", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

//BoxList render
test("BoxList can render", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

//NewBoxListForm render
test("NewBoxListForm list can render", () => {
  const { asFragment } = render(<NewBoxListForm />);
  expect(asFragment()).toMatchSnapshot();
});

//Box render
test("Box list can render", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

//Adding box
test("Adding a box", () => {
  //starting with nothing
  const { container } = render(<BoxList />);
  expect(container.querySelectorAll(".singleBox")).length = 1;

  const wVal = container.querySelector(".width");
  const hVal = container.querySelector(".height");
  const btn = container.querySelector(".subBtn");

  //add one purple
  wVal.value = 20;
  hVal.value = 20;
  fireEvent.click(btn);
  expect(container.querySelectorAll(".singleBox")).length = 2;
  expect(wVal).value = "";
  expect(hVal).value = "";
  expect(container.querySelectorAll(".singleBox")[1]).toHaveStyle(
    `background-color: purple`
  );
  //add one aqua
  wVal.value = 40;
  hVal.value = 20;
  fireEvent.click(btn);
  expect(container.querySelectorAll(".singleBox")).length = 3;
  expect(wVal).value = "";
  expect(hVal).value = "";
  expect(container.querySelectorAll(".singleBox")[2]).toHaveStyle(`width: 40`);
});

//bad input testing
test("should throw errors for bad inputs", () => {
  const { container } = render(<BoxList />);
  const wVal = container.querySelector(".width");
  const hVal = container.querySelector(".height");
  //bad inputs
  expect(() => (wVal = 19)).toThrow("Assignment to constant variable.");
  expect(() => (wVal = 22)).toThrow("Assignment to constant variable.");
  expect(() => (wVal = "test")).toThrow("Assignment to constant variable.");
  expect(() => (hVal = 19)).toThrow("Assignment to constant variable.");
  expect(() => (hVal = 22)).toThrow("Assignment to constant variable.");
  expect(() => (hVal = "test")).toThrow("Assignment to constant variable.");
});
