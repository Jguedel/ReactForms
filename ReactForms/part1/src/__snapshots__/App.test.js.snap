// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App can render 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="boxList"
    >
      <div
        class="boxForm"
      >
        <form>
          <div>
            <label
              for="width"
            >
              Width:
            </label>
            <input
              class="width"
              max="300"
              min="20"
              name="width"
              placeholder="width"
              required=""
              step="20"
              type="number"
              value=""
            />
          </div>
          <div>
            <label
              for="height"
            >
              Height:
            </label>
            <input
              class="height"
              max="300"
              min="20"
              name="height"
              placeholder="height"
              required=""
              step="20"
              type="number"
              value=""
            />
          </div>
          <div>
            <label
              for="color"
            >
              Color:
            </label>
            <select
              class="color"
              id="color"
              name="color"
            >
              <option
                select="true"
                value="purple"
              >
                purple
              </option>
              <option
                value="pink"
              >
                pink
              </option>
              <option
                value="aqua"
              >
                aqua
              </option>
              <option
                value="brown"
              >
                brown
              </option>
            </select>
          </div>
          <button
            class="subBtn"
          >
            Add!
          </button>
        </form>
      </div>
      <div
        class="boxContainer"
      >
        <div
          class="singleBox"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Box list can render 1`] = `
<DocumentFragment>
  <div
    class="singleBox"
  />
</DocumentFragment>
`;

exports[`BoxList can render 1`] = `
<DocumentFragment>
  <div
    class="boxList"
  >
    <div
      class="boxForm"
    >
      <form>
        <div>
          <label
            for="width"
          >
            Width:
          </label>
          <input
            class="width"
            max="300"
            min="20"
            name="width"
            placeholder="width"
            required=""
            step="20"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            for="height"
          >
            Height:
          </label>
          <input
            class="height"
            max="300"
            min="20"
            name="height"
            placeholder="height"
            required=""
            step="20"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            for="color"
          >
            Color:
          </label>
          <select
            class="color"
            id="color"
            name="color"
          >
            <option
              select="true"
              value="purple"
            >
              purple
            </option>
            <option
              value="pink"
            >
              pink
            </option>
            <option
              value="aqua"
            >
              aqua
            </option>
            <option
              value="brown"
            >
              brown
            </option>
          </select>
        </div>
        <button
          class="subBtn"
        >
          Add!
        </button>
      </form>
    </div>
    <div
      class="boxContainer"
    >
      <div
        class="singleBox"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`NewBoxListForm list can render 1`] = `
<DocumentFragment>
  <div
    class="boxForm"
  >
    <form>
      <div>
        <label
          for="width"
        >
          Width:
        </label>
        <input
          class="width"
          max="300"
          min="20"
          name="width"
          placeholder="width"
          required=""
          step="20"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height:
        </label>
        <input
          class="height"
          max="300"
          min="20"
          name="height"
          placeholder="height"
          required=""
          step="20"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="color"
        >
          Color:
        </label>
        <select
          class="color"
          id="color"
          name="color"
        >
          <option
            select="true"
            value="purple"
          >
            purple
          </option>
          <option
            value="pink"
          >
            pink
          </option>
          <option
            value="aqua"
          >
            aqua
          </option>
          <option
            value="brown"
          >
            brown
          </option>
        </select>
      </div>
      <button
        class="subBtn"
      >
        Add!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
